addons/account/models/account_payment.py

```diff
@@ -334,8 +334,8 @@ def _prepare_move_line_default_vals(self, write_off_line_vals=None, force_balanc
         currency_id = self.currency_id.id
 
         # Compute a default label to set on the journal items.
-        liquidity_line_name = ''.join(x[1] for x in self._get_aml_default_display_name_list())
-        counterpart_line_name = ''.join(x[1] for x in self._get_aml_default_display_name_list())
+        liquidity_line_name = ''.join(x[1] or "" for x in self._get_aml_default_display_name_list())
```

Comment:
I would rather check the value.

```suggestion
        liquidity_line_name = ''.join(x[1] for x in self._get_aml_default_display_name_list() if x[1])
```